library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -445,25 +445,37 @@ private long insert(Request request) {
         return ContentUris.parseId(downloadUri);
     }
 
-    public void pauseBatch(long id) {
-        ContentValues values = new ContentValues();
-        String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
-        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
-        values.put(DownloadContract.Downloads.COLUMN_CONTROL, DownloadsControl.CONTROL_PAUSED);
-        contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
-    }
+    public void pauseBatch(long batchId) {
+        DownloadDeleter downloadDeleter = new DownloadDeleter(contentResolver);
+        RealSystemFacade systemFacade = new RealSystemFacade(GlobalState.getContext(), new Clock());
+        BatchRepository batchRepository = new BatchRepository(contentResolver, downloadDeleter, downloadsUriProvider, systemFacade);
 
-    public void resumeBatch(long id) {
-        ContentValues values = getResumeContentValues();
-        String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
-        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
-        contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
+        int batchStatus = batchRepository.getBatchStatus(batchId);
+        if (DownloadStatus.isRunning(batchStatus)) {
+            ContentValues values = new ContentValues();
+            String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
+            String[] selectionArgs = {String.valueOf(batchId), String.valueOf(DownloadStatus.SUCCESS)};
+            values.put(DownloadContract.Downloads.COLUMN_CONTROL, DownloadsControl.CONTROL_PAUSED);
+            contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
+        }
```

Comment:
Exactly, you cannot pause something that is not running :)

On Fri, 24 Jul 2015 16:59 Xavi Rigau <notifications@github.com> wrote:

> In
> library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java
> <https://github.com/novoda/download-manager/pull/135#discussion_r35430263>
> :
>
> >
> > -    public void resumeBatch(long id) {
> > -        ContentValues values = getResumeContentValues();
> > -        String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
> > -        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
> > -        contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
> > +        int batchStatus = batchRepository.getBatchStatus(batchId);
> > +        if (DownloadStatus.isRunning(batchStatus)) {
> > +            ContentValues values = new ContentValues();
> > +            String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
> > +            String[] selectionArgs = {String.valueOf(batchId), String.valueOf(DownloadStatus.SUCCESS)};
> > +            values.put(DownloadContract.Downloads.COLUMN_CONTROL, DownloadsControl.CONTROL_PAUSED);
> > +            contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
> > +        }
>
> do we just ignore ot if the batch is not running?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/download-manager/pull/135/files#r35430263>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93